File: v3/python/newscatcherapi_client/apis/tags/search_api.py
# coding: utf-8

from newscatcherapi_client.apis.tags.search_api_generated import SearchApiGenerated
from newscatcherapi_client.search_all_helper import search_all

class SearchApi(SearchApiGenerated):
    
    def get_all(self, **kwargs):
        return search_all(self.get, **kwargs)

    def post_all(self, **kwargs):
        return search_all(self.post, **kwargs)

File: v3/typescript/api/search-api.ts
import { SearchApiGenerated, SearchApiGetRequest, SearchApiPostRequest } from "./search-api-generated";
import { searchAll } from "../search-all-helper";
import { AxiosRequestConfig } from "axios";
export * from "./search-api-generated";

export class SearchApi extends SearchApiGenerated {

    public async get_all(requestParameters: SearchApiGetRequest, options?: AxiosRequestConfig) {
        return searchAll(this.get.bind(this), requestParameters, options);
    }

    public async post_all(requestParameters: SearchApiPostRequest, options?: AxiosRequestConfig) {
        return searchAll(this.post.bind(this), requestParameters, options);
    }
}
